File: src/games/stendhal/client/events/GroupInviteEvent.java
/* $Id$ */
/***************************************************************************
 *                   (C) Copyright 2003-2010 - Stendhal                    *
 ***************************************************************************
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/
package games.stendhal.client.events;

import games.stendhal.client.entity.RPEntity;
import games.stendhal.client.gui.group.GroupPanelController;

/**
 * The player was invited or the invitation expired
 *
 * @author hendrik
 */
class GroupInviteEvent extends Event<RPEntity> {
	/**
	 * executes the event
	 */
	@Override
	public void execute() {
		if (event.has("expire")) {
			GroupPanelController.get().expireInvite(event.get("leader"));
		} else {
			GroupPanelController.get().receiveInvite(event.get("leader"));
		}
	}
}

File: srcjs/stendhal/event/GroupInviteEvent.ts
/***************************************************************************
 *                 (C) Copyright 2005-2023 - Faiumoni e. V.                *
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Affero General Public License as        *
 *   published by the Free Software Foundation; either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/


import { RPEvent } from "./RPEvent";
import { Chat } from "../util/Chat";

declare var marauroa: any;

/**
 * handles player trade events
 */
export class GroupInviteEvent extends RPEvent {
	public expire!: string;
	public leader!: string;

	public execute(entity: any): void {
		if (entity !== marauroa.me) {
			return;
		}
		if (this["expire"]) {
			Chat.log("normal", "Your group invite by " + this["leader"] + " has expired.");
		} else {
			Chat.log("normal", "Your have been invited by " + this["leader"] + " to join a group.");
			Chat.log("normal", "To join, type: /group join " + this["leader"]);
			Chat.log("normal", "To leave the group at any time, type: /group part " + this["leader"]);
		}
	}
}
